Peer/Apps/App.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using CommandLine;
+using CommandLine.Text;
+using Microsoft.Extensions.DependencyInjection;
+using Peer.Apps.AppBuilder;
+using Peer.Domain.Configuration;
+using wimm.Secundatives;
+
+namespace Peer.Apps;
+
+public class App
+{
+    private readonly ICommandLineParser _parser;
+    private readonly List<IVerb> _verbs;
+    private readonly IServiceSetupHandler? _setupHandler;
+
+    //CN - This is a temp thing - Really we just should fail out if config fails probably.
+    // Alternatively since there's a few different cases where it doesn't matter maybe we don't
+    // care about if it succeeds or fails at this point - or break up the
+    // different handlers.. Since we don't care about loading config till later in 
+    // the process
+    // I think really what we want is: 
+    // AppBuilderTime - Only the stuff required to Create handlers/junk
+    // PreParse - Only the common setup stuff for each handler (Stuff common to everybody - probably don't *need* this)
+    // VerbConfig - All of the stuff required to run specific verbs - initialized right before running it (probably Verb.CustomConfig or similar)
+    private static readonly Dictionary<ConfigError, string> _configErrorMap = new()
+    {
+        [ConfigError.InvalidProviderValues] = "One or more providers have invalid configuration",
+        [ConfigError.NoProvidersConfigured] = "No providers are configured! Run 'peer config' to get started",
+        [ConfigError.ProviderNotMatched] = "Provider was not recognized, make sure you're using one of supported providers!"
```

Comment:
"one of supported providers"

Reply:
good catch (that message has been there for years so bad catch then :P )